packages/compile-solidity/src/run.ts

```diff
@@ -282,7 +348,7 @@ function detectErrors({
   const infos = rawInfos.map(({ formattedMessage }) => formattedMessage);
 
   if (errors.includes("requires different compiler version")) {
-    const contractSolcVer = errors.match(/pragma solidity[^;]*/gm)[0];
+    const contractSolcVer = errors.match(/pragma solidity[^;]*/gm)![0];
```

Comment:
I don't think the use of the `!` operator is safe here. I'd recommend using a type assertion instead.

Reply:
So in a lot of these situations I just tried to maintain current behavior. You think I should make it throw? I suppose that would be an improvement.

Reply:
Yeah, it throws either way if the `!` is invalid. If you throw on a null assertion, at least you can provide some context.

Otherwise if we leave it strictly identical in behavior in cases like this, we're negating the primary value that the TS type checker provides, IMO.

Reply:
Also an explicit throw is more readable, IMO, as that makes it a lot more clear that this code _can_ throw.